libbpfgo.go

```diff
@@ -578,6 +578,116 @@ func (b *BPFMap) GetValue(key unsafe.Pointer) ([]byte, error) {
 	return value, nil
 }
 
+// BPFMapBatchOpts mirrors the C structure bpf_map_batch_opts.
+type BPFMapBatchOpts struct {
+	Sz        uint64
+	ElemFlags uint64
+	Flags     uint64
+}
+
+func bpfMapBatchOptsToC(batchOpts *BPFMapBatchOpts) *C.struct_bpf_map_batch_opts {
+	if batchOpts == nil {
+		return nil
+	}
+	opts := C.struct_bpf_map_batch_opts{}
+	opts.sz = C.ulong(batchOpts.Sz)
+	opts.elem_flags = C.ulonglong(batchOpts.ElemFlags)
+	opts.flags = C.ulonglong(batchOpts.Flags)
+
+	return &opts
+}
+
+// GetValueBatch allows for batch lookups of multiple keys.
+// The first argument is a pointer to an array of keys which will be populated with the keys returned from this operation.
+// It returns the associated values as a slice of slices of bytes.
```

Comment:
That SGTM, will update.

Reply:
Actually, the way I would implement this would be to return `map[[]byte][]byte` because we don't know what type the keys will be (or size). The problem with this is a `[]byte` is not comparable so cannot be used as a key. I suppose we could use a `map[interface{}][]byte` and that should work (haven't tested it) but feels kind of weird to me, and adds a type assertion to the callers code.

I'm happy to implement the latter if that's what you'd prefer, however this interface matches what happens in libbpf (although admittedly it's not very Go-like).

As shown in the test it's fairly straightforward still to iterate through keys / values using a single iterator var to keep track of where you are.

Reply:
Good points, best to leave it as is.

Reply:
Sounds good!